react-components/src/desktop/view/Taskbar.js

```diff
@@ -0,0 +1,28 @@
+/**
+ * @author sriram
+ */
+import React, { Component } from "react";
+import styles from "../style";
+import TaskButton from "./TaskButton";
+import Toolbar from "@material-ui/core/Toolbar";
+import { withStyles } from "@material-ui/core/styles";
+
+
+class Taskbar extends Component {
+    render() {
+        const {windows, classes} = this.props;
+        return (
+            <Toolbar className={classes.taskbar}>
+                {windows.map(n => {
+                    return (
+                        <TaskButton windowConfig={n}
```

Comment:
I just noticed that a `key` prop is getting added to the inner `Button` returned by `TaskButton`, but that is not the appropriate place to set a `key` prop of an item in an array (it's possible for `TaskButton` to be used on its own outside an array of children).

It should be set here instead, where the array is constructed by the parent component:
```jsx
                        <TaskButton key={n.tag} 
                                    windowConfig={n}
                                    taskButtonClickHandler={this.props.taskButtonClickHandler}/>
```

Reply:
👍 I was wondering why I got warnings even though I had the keys. Good catch.